Add information on class isolation for JDBC driver (#629)
tableau/connector-plugin-sdk: docs/design.md

For information on common capabilities and how they are used, see [Capabilities]({{ site.baseurl }}/docs/capabilities).

## JDBC Driver Class Isolation
If the driver only includes a single jar file, copy it to the JDBC driver location. <br/>
If the driver includes more than a single file, create a unique subfolder under JDBC driver location and include all required files. 
- Windows: C:\Program Files\Tableau\Drivers
- Mac: ~/Library/Tableau/Drivers
- Linux: /opt/tableau/tableau_driver/jdbc 
<br/> 

> Copy all files to folder "A". This will create an isolated classloader for that driver. To learn more about JDBC drivers, check *Specify the right JDBC driver* section in  the  [user documentation](https://help.tableau.com/current/pro/desktop/en-us/examples_otherdatabases_jdbc.htm).

Reviewer: ```suggestion
To learn more about JDBC drivers, check *Specify the right JDBC driver* section in  the  [user documentation](https://help.tableau.com/current/pro/desktop/en-us/examples_otherdatabases_jdbc.htm).
```
"This will create an isolated classloader for that driver. " makes more sense on line 89 I think.

Author: Got it. Will add that.

For information on common capabilities and how they are used, see [Capabilities]({{ site.baseurl }}/docs/capabilities).

> ## JDBC Driver Class Isolation

Reviewer: Is there any user documentation or a KB article that also includes this information.  The Connector SDK should be a supporting document and not the definitive source for this information.  If so can we link to it.

Author: Will add the link to the user documentation

> If the driver only includes a single jar file, just copy it to the JDBC driver loaction. <br/>

Reviewer: ```suggestion
If the driver only includes a single jar file, copy it to the JDBC driver location. <br/>
```

> If the driver includes more than a single file, create a subfolder "A" under JDBC driver location. 

Reviewer: ```suggestion
If the driver includes more than a single file, create a unique subfolder under JDBC driver location and include all required files. 
```
